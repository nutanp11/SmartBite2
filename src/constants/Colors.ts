export const Colors = {
    appBackground: '#ffffff',
    navyBlue: '#274d76',
    darkGrey: '#333333',
    grey: '#999b9d',
    white: '#FFFFFF',
    errorRed: '#eb0708',
    fontBlack: '#373b41',
    fadedBlue: '#5d96ba',
    blue: '#5d96ba',
    lightGrey: '#bcbcbc',
    mediumGrey: '#787a7c',
    fadedGrey: '#e0e0e0',
    yellow: '#f3cf52',
    stickiesBlues: '#ecf3f8',
    lightBlue: '#F0FFFF',
    goldenYellow: '#FFB300',
    black: '#111111',
    lightGreen: '#41DE00',
    green: '#66C03D',
    red: '#EA1B1A',
    lightGreyShade: '#F6F6F6',
    succussColor: '#66C03D',
    successGreenBackground: '#EEFFDE',
    darkGreen: '#002C00',
    textGray: '#999999',
    appTheme: '#2F2E41',
    grayBack: '#F4F4F4',
    extrablack: '#000000',
    tabBackground: '#F2F5FD',
    borderColor: '#F0F3FB',
    appDarkTheme: '#100F18'
  };
 