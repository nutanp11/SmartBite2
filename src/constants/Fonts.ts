export const Fonts = {
regular : 'Nunito-Regular',
bold: 'Nunito-Bold',
semibold: 'Nunito-SemiBold',
NunitoMedium: 'Nunito-Medium',
NunitoMediumItalic: 'Nunito-MediumItalic',
NunitoExtraBold: 'Nunito-ExtraBold',
};