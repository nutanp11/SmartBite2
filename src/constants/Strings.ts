export const SEARCH_TEXT = 'What do you looking for?';
export const WELCOME_TEXT = 'Welcome Back, ';
export const ENTER_INFO = 'Please type your information';
export const ENTER_EMAIL = 'Email address';
export const PASSWORD = 'Password';
export const NEW_USER = 'New user? ';
export const CREATE_ACCOUNT = 'Create an account';
export const ALREADY_ACCOUNT = 'Already have an account?';
export const LOG_IN = 'Login';
export const FULL_NAME = "Full name";
export const CREATE_ACC_BTN = 'Create Account'