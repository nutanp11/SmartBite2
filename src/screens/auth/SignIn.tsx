//SignIn.tsx
/* eslint-disable react-native/no-inline-styles */
import {
    StyleSheet,
    Text,
    View,
    Image,
    ScrollView,
    TextStyle,
    ViewStyle,
    ImageStyle,
    Alert,
    ImageBackground,
} from 'react-native';
import React, { useState } from 'react';
import { Colors } from '../../constants/Colors';
import { Fonts } from '../../constants/Fonts';
import AppButton from '../../components/AppButton';

import TextInputComp from '../../components/TextInputComp';
import HeaderText from '../../components/HeaderText';
import {
    CREATE_ACCOUNT,
    ENTER_EMAIL,
    ENTER_INFO,
    LOG_IN,
    NEW_USER,
    PASSWORD,
} from '../../constants/Strings';
import { width } from '../../utils/comman';

type SignInNavigationProp = {
    navigate: (screen: string) => void;
};

type SignInProps = {
    navigation: SignInNavigationProp;
};

const SignIn: React.FC<SignInProps> = ({ navigation }) => {
    const [email, setEmail] = useState<string>('');

    const [password, setPassword] = useState<string>('');

    const validateForm = () => {
        let isValid = true;
        if (email.trim() === '') {
            isValid = false;
        }
        if (password.trim() === '') {
            isValid = false;
        }
        return isValid;
    };

    const onPressLogin = () => {
        const isValid = validateForm();
        if (isValid) {
            Alert.alert('Form Submitted', 'Your email and password are valid.');
            navigation.navigate('ScreenA');
        } else {
            Alert.alert('Validation Failed', 'Please fill in all fields correctly.');
        }
    };

    return (
        <ScrollView contentContainerStyle={{ flexGrow: 1 }}>
            <View style={styles.container}>
                <View style={styles.topview}>
                    <View >
                        <ImageBackground style={{ width: width, height: width / 1.2,alignItems:"center",justifyContent:"center",
                             borderBottomLeftRadius: 10, alignSelf: 'center' }} resizeMode='cover' source={{ uri: "https://th.bing.com/th/id/OIP.QRlnlD0sHv_Ii_ExO3QwhwHaEo?w=316&h=197&c=8&rs=1&qlt=90&o=6&dpr=1.5&pid=3.1&rm=2" }} >
                        <HeaderText title={'SmartBite'} />

                        </ImageBackground>

                        <Text style={styles.loginText}>India's #1 Food Delivery and Dining App</Text>

                    </View>

                    <View style={{  justifyContent:"space-around",top:20
 }}>

                        <TextInputComp
                            value={email}
                            placeholderText={ENTER_EMAIL}
                            onChangeText={(text: string) => setEmail(text)}
                        />

                        <TextInputComp
                            value={password}
                            placeholderText={PASSWORD}
                            onChangeText={(text: string) => setPassword(text)}
                        />
                    </View>

                </View>

                <View style={styles.bottomview}>
                    <AppButton onPress={onPressLogin} title={LOG_IN} />

                </View>
            </View>
        </ScrollView>
    );
};

export default SignIn;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: Colors.appBackground,
    } as ViewStyle,
    topview: {
        flex: 0.8,
        padding: 10,
    } as ViewStyle,
    bottomview: {
        flex: 0.2,
        justifyContent: 'space-between',
    } as ViewStyle,
    loginText: {
        fontSize: 26,
        fontWeight: '900',
        fontFamily: Fonts.bold,
        alignSelf: 'center',
        textAlign: "center",
        width: '80%',

    } as TextStyle,
    subtext: {
        fontSize: 14,
        fontWeight: '600',
        lineHeight: 35,
        color: Colors.textGray,
        fontFamily: Fonts.regular,
        alignSelf: 'center',
    } as TextStyle,
    imgstyle: {
        width: 256,
        height: 251,
        marginVertical: 10,
        alignSelf: 'center',
    } as ImageStyle,
    graytext: {
        fontSize: 14,
        fontWeight: '600',
        color: Colors.textGray,
        fontFamily: Fonts.regular,
        alignSelf: 'center',
    } as TextStyle,
    blacktext: {
        fontSize: 14,
        fontFamily: Fonts.semibold,
    } as TextStyle,
    viewStyle: {
        flexDirection: 'row',
        alignSelf: 'center',
    } as ViewStyle,
});
