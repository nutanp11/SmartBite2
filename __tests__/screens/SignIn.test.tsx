import { fireEvent, render, waitFor } from '@testing-library/react-native';
import SignIn from '../../src/screens/auth/SignIn';
import { ENTER_EMAIL, LOG_IN, PASSWORD } from '../../src/constants/Strings';

describe('SignIn Component', () => {
  test('should render email and password input fields', () => {
    const { getByPlaceholderText } = render(
      <SignIn navigation={{ navigate: jest.fn() }} />
    );

    expect(getByPlaceholderText(ENTER_EMAIL)).toBeTruthy();
    expect(getByPlaceholderText(PASSWORD)).toBeTruthy();
  });

  test('should handle email input change', () => {
    const { getByPlaceholderText } = render(
      <SignIn navigation={{ navigate: jest.fn() }} />
    );

    const emailInput = getByPlaceholderText(ENTER_EMAIL);
    fireEvent.changeText(emailInput, 'test@example.com');
    expect(emailInput.props.value).toBe('test@example.com');
  });

  test('should handle password input change', () => {
    const { getByPlaceholderText } = render(
      <SignIn navigation={{ navigate: jest.fn() }} />
    );

    const passwordInput = getByPlaceholderText(PASSWORD);
    fireEvent.changeText(passwordInput, 'password123');
    expect(passwordInput.props.value).toBe('password123');
  });

  test('should trigger navigation on valid input', async () => {
    const mockNavigate = jest.fn();
    const { getByPlaceholderText, getByText } = render(
      <SignIn navigation={{ navigate: mockNavigate }} />
    );

    fireEvent.changeText(getByPlaceholderText(ENTER_EMAIL), 'test@example.com');
    fireEvent.changeText(getByPlaceholderText(PASSWORD), 'password123');
    fireEvent.press(getByText(LOG_IN));

    await waitFor(() => {
      expect(mockNavigate).toHaveBeenCalledWith('Dashboard');
    });
  });

  test('should show error when email is empty', async () => {
    const { getByPlaceholderText, getByText } = render(
      <SignIn navigation={{ navigate: jest.fn() }} />
    );

    fireEvent.changeText(getByPlaceholderText(ENTER_EMAIL), '');
    fireEvent.changeText(getByPlaceholderText(PASSWORD), 'password123');
    fireEvent.press(getByText(LOG_IN));

    await waitFor(() => {
      expect(getByText('Email is required')).toBeTruthy();
    });
  });

  test('should show error when password is too short', async () => {
    const { getByPlaceholderText, getByText } = render(
      <SignIn navigation={{ navigate: jest.fn() }} />
    );

    fireEvent.changeText(getByPlaceholderText(ENTER_EMAIL), 'test@example.com');
    fireEvent.changeText(getByPlaceholderText(PASSWORD), '123');
    fireEvent.press(getByText(LOG_IN));

    await waitFor(() => {
      // Wait for error text to appear
      expect(getByText('Password must be at least 6 characters')).toBeTruthy();
    });
  });

  test('should show error when both fields are empty', async () => {
    const { getByPlaceholderText, getByText } = render(
      <SignIn navigation={{ navigate: jest.fn() }} />
    );

    fireEvent.changeText(getByPlaceholderText(ENTER_EMAIL), '');
    fireEvent.changeText(getByPlaceholderText(PASSWORD), '');
    fireEvent.press(getByText(LOG_IN));

    await waitFor(() => {
      // Wait for error texts to appear
      expect(getByText('Email is required')).toBeTruthy();
      expect(getByText('Password is required')).toBeTruthy();
    });
  });
});
