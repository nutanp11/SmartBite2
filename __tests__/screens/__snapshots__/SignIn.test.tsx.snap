// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#ffffff",
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "flex": 0.8,
            "padding": 10,
          }
        }
      >
        <View>
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              {
                "alignSelf": "center",
                "borderBottomLeftRadius": 10,
                "height": 625,
                "top": -10,
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="stretch"
              source={
                {
                  "uri": "https://th.bing.com/th/id/OIP.OHv_QRoJDtQwoqYTrkKEmwHaE6?w=277&h=184&c=7&r=0&o=5&dpr=1.5&pid=1.7",
                }
              }
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  {
                    "height": 625,
                    "width": 750,
                  },
                  undefined,
                ]
              }
            />
            <View>
              <Text
                style={
                  [
                    {
                      "alignSelf": "center",
                      "fontFamily": "Nunito-Bold",
                      "fontSize": 54,
                      "fontWeight": "900",
                      "marginTop": 0,
                    },
                    {
                      "color": "#FFFFFF",
                      "fontSize": 38,
                      "top": 40,
                    },
                  ]
                }
              >
                SmartBite
              </Text>
            </View>
          </View>
          <Text
            style={
              {
                "alignSelf": "center",
                "fontFamily": "Nunito-Bold",
                "fontSize": 26,
                "fontWeight": "900",
                "textAlign": "center",
                "top": 10,
                "width": "80%",
              }
            }
          >
            India's #1 Food Delivery and Dining App
          </Text>
        </View>
        <View
          style={
            {
              "justifyContent": "space-around",
            }
          }
        >
          <View>
            <Text
              style={
                [
                  {
                    "alignSelf": "center",
                    "fontFamily": "Nunito-Bold",
                    "fontSize": 54,
                    "fontWeight": "900",
                    "marginTop": 0,
                  },
                  {
                    "color": "#111111",
                    "fontSize": 32,
                    "marginVertical": 10,
                  },
                ]
              }
            >
              Login
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "center",
                  "backgroundColor": "#F4F4F4",
                  "borderRadius": 5,
                  "marginVertical": 10,
                  "padding": 5,
                  "width": "90%",
                },
                undefined,
              ]
            }
          >
            <TextInput
              keyboardType="default"
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Email address"
              placeholderTextColor="#000000"
              style={
                {
                  "color": "#000000",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 14,
                  "fontWeight": "900",
                  "padding": 15,
                }
              }
              value=""
            />
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "center",
                  "backgroundColor": "#F4F4F4",
                  "borderRadius": 5,
                  "marginVertical": 10,
                  "padding": 5,
                  "width": "90%",
                },
                undefined,
              ]
            }
          >
            <TextInput
              keyboardType="default"
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="#000000"
              secureTextEntry={true}
              style={
                {
                  "color": "#000000",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 14,
                  "fontWeight": "900",
                  "padding": 15,
                }
              }
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flex": 0.2,
            "justifyContent": "space-between",
          }
        }
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#DE4B4C",
                "borderRadius": 10,
                "opacity": 1,
                "padding": 10,
                "width": "80%",
              }
            }
          >
            <Text
              style={
                {
                  "alignSelf": "center",
                  "color": "#FFFFFF",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 24,
                  "fontWeight": "900",
                  "lineHeight": 45,
                }
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
